community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/QueryGraph.scala

```diff
@@ -36,6 +36,66 @@ case class QueryGraph(patternRelationships: Set[PatternRelationship] = Set.empty
                       mutatingPatterns: Seq[MutatingPattern] = Seq.empty)
   extends UpdateGraph {
 
+  def smallestGraphIncluding(mustInclude: Set[IdName]): Set[IdName] = {
```

Comment:
That would be really difficult. I think I see what you mean. Let's keep that option open in the future